ui/page/governance/agenda_vote_modal.go

```diff
@@ -0,0 +1,307 @@
+		Load:              l,
+		modal:             *l.Theme.ModalFloatTitle(),
+		agenda:            agenda,
+		consensusPage:     consensusPage,
+		materialLoader:    material.Loader(material.NewTheme(gofont.Collection())),
+		optionsRadioGroup: new(widget.Enum),
+		spendingPassword:  l.Theme.EditorPassword(new(widget.Editor), "Spending password"),
+		voteBtn:           l.Theme.Button("Update Preference"),
+		cancelBtn:         l.Theme.OutlineButton("Cancel"),
+	}
+
+	avm.voteBtn.Background = l.Theme.Color.Gray3
+	avm.voteBtn.Color = l.Theme.Color.Surface
+
+	avm.walletSelector = NewWalletSelector(l).
+		Title("Voting wallet").
+		WalletSelected(func(w *dcrlibwallet.Wallet) {
+			avm.loadCount = 0
+
+			avm.FetchUnspentUnexpiredTickets(w.ID)
+			avm.RefreshWindow()
+			avm.loadCount++
+
+			// update agenda options prefrence to that of the selected wallet
+			consensusItems := components.LoadAgendas(avm.Load, w, false)
+			for _, consensusItem := range consensusItems {
+				if consensusItem.Agenda.AgendaID == agenda.AgendaID {
+					ArrVoteOptions := make(map[string]string)
+					for i := range consensusItem.Agenda.Choices {
+						ArrVoteOptions[agenda.Choices[i].Id] = consensusItem.Agenda.Choices[i].Id
+					}
+
+					// sort keys to keep order when refreshed
+					sortedKeys := make([]string, 0)
+					for k := range ArrVoteOptions {
+						sortedKeys = append(sortedKeys, k)
+					}
+
+					sort.Slice(sortedKeys, func(i, j int) bool {
+						var stringI string = sortedKeys[i]
+						var stringJ string = sortedKeys[j]
+						if stringI == stringJ {
+							return stringI < stringJ
+						}
+						return stringI < stringJ
+					})
+
+					avm.itemKeys = sortedKeys
+					avm.items = ArrVoteOptions
+
+					initialValue := consensusItem.Agenda.VotingPreference
+					if initialValue == "" {
+						initialValue = avm.defaultValue
+					}
+
+					avm.initialValue = initialValue
+					avm.currentValue = initialValue
+
+					avm.optionsRadioGroup.Value = avm.currentValue
+				}
+			}
+		}).
+		WalletValidator(func(w *dcrlibwallet.Wallet) bool {
+			return !w.IsWatchingOnlyWallet()
+		})
+
+	return avm
+}
+
+func (avm *agendaVoteModal) FetchUnspentUnexpiredTickets(walletID int) {
+	go func() {
+		wallet := avm.WL.MultiWallet.WalletWithID(walletID)
+		tickets, err := wallet.UnspentUnexpiredTickets()
+		if err != nil {
+			avm.Toast.NotifyError(err.Error())
+			return
+		}
+
+		liveTickets := make([]*dcrlibwallet.Transaction, 0)
+		txItems, err := components.StakeToTransactionItems(avm.Load, tickets, true, func(filter int32) bool {
+			switch filter {
+			case dcrlibwallet.TxFilterUnmined:
+				fallthrough
+			case dcrlibwallet.TxFilterImmature:
+				fallthrough
+			case dcrlibwallet.TxFilterLive:
+				return true
+			}
+
+			return false
+		})
+		if err != nil {
+			avm.Toast.NotifyError(err.Error())
+			return
+		}
+
+		for _, liveTicket := range txItems {
+			liveTickets = append(liveTickets, liveTicket.Transaction)
+		}
+
+		avm.LiveTickets = liveTickets
```

Comment:
You don't need these. It seems the `StakeToTransactionItems` was designed to deal with voted and expired tickets which we don't have here. Only other sensible thing it does is sort.
```suggestion
		// sort by newest first
		sort.Slice(tickets[:], func(i, j int) bool {
			var timeStampI = tickets[i].Timestamp
			var timeStampJ = tickets[j].Timestamp
			return timeStampI > timeStampJ
		})
		avm.LiveTickets = make([]*dcrlibwallet.Transaction, len(tickets))
		for i := range tickets {
			avm.LiveTickets[i] = &tickets[i]
		}
```